File: hospital.client/src/types/NotificationLogResponse.ts
export interface NotificationLogResponse {
  id: number;
  recipientEmail: string;
  subject: string;
  notificationType: number;
  relatedEntityType: string;
  relatedEntityId?: number | null;
  sentAt?: string | null;
  status: number;
  retryCount?: number | null;
  errorMessage?: string | null;
  state: number;
  createdAt: string;
  createdBy: number;
  updatedBy?: number | null;
  updatedAt?: string | null;
}

export interface NotificationLogRequest {
  id?: number | null;
  recipientEmail?: string | null;
  subject?: string | null;
  notificationType?: number | null;
  relatedEntityType?: string | null;
  relatedEntityId?: number | null;
  sentAt?: string | null;
  status?: number | null;
  retryCount?: number | null;
  errorMessage?: string | null;
  state?: number | null;
  createdBy?: number | null;
  updatedBy?: number | null;
}

File: Hospital.Server/Entities/Response/NotificationLogResponse.cs
namespace Hospital.Server.Entities.Response
{
    public class NotificationLogResponse
    {
        public long Id { get; set; }
        public string RecipientEmail { get; set; } = string.Empty;
        public string Subject { get; set; } = string.Empty;
        public int NotificationType { get; set; }
        public string RelatedEntityType { get; set; } = string.Empty;
        public long RelatedEntityId { get; set; }
        public string? SentAt { get; set; }
        public int Status { get; set; }
        public int RetryCount { get; set; }
        public string? ErrorMessage { get; set; }
        public int State { get; set; }
        public string CreatedAt { get; set; } = string.Empty;
        public long CreatedBy { get; set; }
        public long? UpdatedBy { get; set; }
        public string? UpdatedAt { get; set; }
    }
}
